absl/strings/str_cat.cc

```diff
@@ -89,7 +89,7 @@ static char* Append(char* out, const AlphaNum& x) {
   // memcpy is allowed to overwrite arbitrary memory, so doing this after the
   // call would force an extra fetch of x.size().
   char* after = out + x.size();
-  if (x.size() != 0) {
+  if (ABSL_PREDICT_TRUE(x.size() != 0)) {
```

Comment:
My gut reaction is that `ABSL_PREDICT_TRUE` is a premature optimization here. Passing an empty string is a legitimate use of `absl::StrCat`. We also have no evidence yet that the branch predictor is getting this wrong. We also use profile-guided optimization in production binaries. This use of `ABSL_PREDICT_TRUE` will prevent the branch predictor from doing its job.